### ISSUE TITLE
[Question] Implementation detail in lime_tabluar/explain_instance

### ISSUE THREAD

**Issue by olgakravchenko** [marcotcr/lime#367]
I am planning to use parts of the LIME code in different parts of a custom ML pipeline, and have hence been looking at how things are implemented. There's an implementation detail that I do not quite understand, namely: 
`distances = sklearn.metrics.pairwise_distances(
                scaled_data,
                scaled_data[0].reshape(1, -1),
                metric=distance_metric
        ).ravel()`
(line 346).  Scaled data here is a scaled version of data:
`data, inverse = self.__data_inverse(data_row, num_samples)`
(line 337). The way the code works now is for categorical variables, all the rows in data are binary, corresponding to feature being "on" or "off". Could you clarify why the distance is calculated for those, and not for inverse variable? The original paper talks about the distance function between the datapoints in the original space, hence my confusion. Thank you.

**Comment by marcotcr** [comment]
The original paper does not contemplate tabular data (numerical and categorical variables). It is easier to come up with distances in the original space that are somewhat robust to different applications for text and images, but I think it's a little harder for tabular data. Not using scaled data would be weird for non-ordinal categorical features that are encoded as integers, as the distance between "1" and "3" should be the same as "1" and "2" (e.g. assume the feature is country, and 1=USA, 2=Brazil, 3=UK). It would also make it much harder to come up with default thresholds for numerical features, as the distances would not be comparable for different domains.
